xbmc/network/websocket/WebSocket.cpp

```diff
@@ -242,7 +242,7 @@ CWebSocketFrame::~CWebSocketFrame()
 
   if (m_free && m_data != NULL)
   {
-    delete m_data;
+    delete[] m_data;
```

Comment:
If you don't mind, I'd like to keep it as is.
m_data has a double role in this file, in one case it's a plain const char* that get's a parameter assigned and is not freed (see m_free flag) - from a quick look I can't get why it's done like that, but I guess there's a reason.
In other cases it's a variable sized array on the heap and needs to be freed.